### ISSUE TITLE
Serving queries to users can be blocked by heavy algorithm computation

### ISSUE THREAD

**Issue by stsievert** [stsievert/salmon#35]
This isn't an issue for light computation by the algorithms, like `RandomSampling`. It's done on a background thread, so it's not that bad.

It'd be nice to have some computational manager to handle heavy computation that algorithms may require.

**Comment by stsievert** [comment]
Some relevant notes from the documentation:

* The central scheduler spends a few hundred microseconds on every task. For optimal performance, task durations should be greater than 10-100ms. ([source][1])
* The client should use `Client(asynchronous=True)`. This has been used to integrate Dask into other asynchronous services (such as web backends), supplying a computational engine similar to Celery, but while still maintaining a high degree of concurrency and not blocking needlessly. ([source][2])
* In production you may wish to run dask-workers within containers ([source][1]).
* Dask enables the remote execution of arbitrary code. You should only host dask-workers within networks that you trust. ([source][1]) (read: the Dask port should use `expose` not `ports`).

[1]:https://distributed.dask.org/en/latest/limitations.html
[2]:https://distributed.dask.org/en/latest/asynchronous.html

**Comment by stsievert** [comment]
I also need to rethink the `Runner.run` function. Here's my current though: update the model and run the search concurrently. Something like following:

``` python
answers = []
while True:
    answers = self.get_answers(rj, clear=True)
    queries_scores, new_embedding = asyncio.gather(self.get_queries(), self.process_answers(answers))
    if self.clear:
        self.clear_queries()
    if len(queries_scores[0]):
        self.post_queries(*queries_scores)
```

Currently, the answer processing and query searching is done serially. Of course, with the addition of concurrency I'll need to be careful that the embedding is constant the entire time.

**Comment by stsievert** [comment]
Dask is the wrong tool for this. I need the following:

* Profiling/dashboard
* Running (a long running) job on separate process/core
